controllers/etcd_controller.go

```diff
@@ -1123,11 +1199,63 @@ func (r *EtcdReconciler) waitUntilStatefulSetReady(sts *appsv1.StatefulSet) (*ap
 		if err := health.CheckStatefulSet(ss); err != nil {
 			return gardenerretry.MinorError(err)
 		}
+		if ss.Status.Replicas != *ss.Spec.Replicas {
+			return gardenerretry.MinorError(fmt.Errorf("ss not scaled"))
+		}
 		return gardenerretry.Ok()
 	})
 	return ss, err
 }
 
+func (r *EtcdReconciler) prepareEtcdForScaleDown(etcd *druidv1alpha1.Etcd, ss *appsv1.StatefulSet, svc *v1.Service) (*corev1.PersistentVolumeClaimList, error) {
+	// TODO: wait for etcd health to be false so that etcd is not serving traffic
+
+	if err := r.takeFullSnapshot(etcd, svc); err != nil {
+		return nil, err
+	}
+
+	selector, err := metav1.LabelSelectorAsSelector(ss.Spec.Selector)
```

Comment:
Isn't statefulset identifies PVC by name?

Reply:
Getting PVC by labels seemed safer than getting it by name.